clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1087,6 +1087,10 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
 
     if (SYCLNativeCPUBackend) {
       llvm::sycl::utils::addSYCLNativeCPUBackendPasses(MPM, MAM, Level);
+      // Run optimization passes after all the changes we made to the kernels.
+      // Todo: maybe we could find a set of relevant passes instead of re-running
+      // the full optimization pipeline.
+      MPM.addPass(PB.buildPerModuleDefaultPipeline(Level));
```

Comment:
Minor (Does not block the PR): It could perhaps be helpful to mention in the comment that the invocation of the pipeline had to be moved into core clang because in the original place in `SYCLLowerIR` it introduces a circular dependency that causes problems when building as shared library.